src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurRequestExtractFileValidationServiceImpl.java

```diff
@@ -0,0 +1,76 @@
+package edu.cornell.kfs.concur.batch.service.impl;
+
+import java.util.List;
+
+import edu.cornell.kfs.concur.batch.businessobject.ConcurRequestExtractFile;
+import edu.cornell.kfs.concur.batch.businessobject.ConcurRequestExtractRequestDetailFileLine;
+import edu.cornell.kfs.concur.batch.businessobject.ConcurRequestExtractRequestEntryDetailFileLine;
+import edu.cornell.kfs.concur.batch.service.ConcurRequestExtractFileValidationService;
+
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+
+public class ConcurRequestExtractFileValidationServiceImpl implements ConcurRequestExtractFileValidationService {
+
+    private static org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(ConcurRequestExtractFileValidationServiceImpl.class);
+
+    public boolean fileRowCountMatchesHeaderRowCount(ConcurRequestExtractFile requestExtractFile) {
+        int fileLineCount = getTotalRequestFileRowCount(requestExtractFile);
+        if (fileLineCount == requestExtractFile.getRecordCount()) {
+            return true;
+        }
+        else {
+            LOG.error("Header row count validation failed. Header record count was =" + requestExtractFile.getRecordCount() + "= while file line count was =" + fileLineCount + "=");
+            return false;
+        }
+    }
+
+    public boolean fileApprovedAmountsMatchHeaderApprovedAmount(ConcurRequestExtractFile requestExtractFile) {
+        KualiDecimal detailLinesAmountSumKualiDecimal = getTotalRequestFileRequestAmount(requestExtractFile);
+        if (detailLinesAmountSumKualiDecimal.doubleValue() == requestExtractFile.getTotalApprovedAmount().doubleValue()) {
```

Comment:
I am not sure why the comparison between the doubleValue is needed here. You can use the KualiDecimal equals or compareTo methods for this test. (equals is calling the compareTo)